versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CachingPersistImpl.java

```diff
@@ -48,14 +48,9 @@ public Obj fetchObj(@Nonnull @jakarta.annotation.Nonnull ObjId id) throws ObjNot
     if (o != null) {
       return o;
     }
-    try {
-      o = persist.fetchObj(id);
-      cache.put(o);
-      return o;
-    } catch (ObjNotFoundException e) {
-      cache.remove(id);
-      throw e;
-    }
+    o = persist.fetchObj(id);
+    cache.put(o);
```

Comment:
Thinking more about upserts and concurrent access, I'm not sure the impl. is perfectly correct. I believe with concurrent get/upsert we can end up with inconsistent data in the cache even without failures. WDYT?

Reply:
Again - upsert is _not_ meant for production use cases - only for repository import - see [javadoc](https://github.com/projectnessie/nessie/blob/f8d41bc8bcd0434de88111da1552d95966a52843/versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/persist/Persist.java#L293-L294).